metacat-common/src/main/java/com/netflix/metacat/common/QualifiedName.java

```diff
@@ -291,6 +298,47 @@ public static QualifiedName ofTable(
         return new QualifiedName(catalogName, databaseName, tableName, null, null);
     }
 
+    private static QualifiedName createCaseSensitiveQualifiedNameIfNecessary(
+        @NonNull final String catalogName,
+        @Nullable final String databaseName,
+        @NonNull final String standardDatabaseName,
+        @Nullable final String tableName,
+        @NonNull final String standardTableName,
+        @Nullable final String partitionName,
+        @NonNull final String standardPartitionName,
+        @Nullable final String viewName,
+        @NonNull final String standardViewName,
+        @NonNull final Type type
+    ) {
+        final String caseSensitiveDatabaseName = standardizeOptional(databaseName, false);
+        final String caseSensitiveTableName = standardizeOptional(tableName, false);
+        final String caseSensitivePartitionName = standardizeOptional(partitionName, false);
+        final String caseSensitiveViewName = standardizeOptional(viewName, false);
+        if (shouldCreateCaseSensitiveQualifiedName(caseSensitiveDatabaseName, standardDatabaseName,
+            caseSensitiveTableName, standardTableName, caseSensitivePartitionName, standardPartitionName,
+            caseSensitiveViewName, standardViewName)) {
+            return new QualifiedName(catalogName, caseSensitiveDatabaseName, caseSensitiveTableName,
+                caseSensitivePartitionName, caseSensitiveViewName, type);
+        }
+
+        return null;
+    }
+
+    private static boolean shouldCreateCaseSensitiveQualifiedName(
+        @NonNull final String databaseName,
+        @NonNull final String standardDatabaseName,
+        @NonNull final String tableName,
+        @NonNull final String standardTableName,
+        @NonNull final String partitionName,
+        @NonNull final String standardPartitionName,
+        @NonNull final String viewName,
+        @NonNull final String standardViewName) {
+        return (!databaseName.isEmpty() && !standardDatabaseName.equals(databaseName))
```

Comment:
Do we need the isEmpty check?

Reply:
Ah not anymore, was using the NonStandardized string earlier. Removed.